Dockerfile

```diff
@@ -1,19 +1,42 @@
 FROM python:alpine3.10
 
-RUN mkdir -p /usr/src/app
-RUN mkdir -p /usr/src/openapi
+RUN apk add --no-cache \
+    gcc \
+    musl-dev \
+    linux-headers \
+    nginx
+
+RUN mkdir -p /usr/src/app /usr/src/openapi /var/log/supervisor /etc/supervisord.d /opt/logs
 WORKDIR /usr/src/app
 
+
+
 COPY requirements.txt /usr/src/app/
 
 ## install the modules
 RUN python3 -m pip install --upgrade pip
 RUN pip3 install --no-cache-dir -r requirements.txt
+RUN pip3 install supervisor
+
+COPY /config/supervisord /etc/rc.d/init.d/supervisord
+ADD  /config/services/* /etc/supervisord.d/
+COPY /config/nginx/nginx.conf /etc/nginx/nginx.conf
```

Comment:
Do we need nginx for uwsgi to work. 
1. We are only exposing API (no static content)
2. We would have a common gateway/bff/nginx in-front of these micro-API.

could you check if just with uwsgi serviing as a web server is sufficient.

Reply:
we require nginx to route the request from container's port to the wsgi application. nginx will constantly listen to a given port and on the basis of paths defined will route the request to appropriate app.